print.c

```diff
@@ -0,0 +1,132 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   print.c                                            :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: nkhribec <marvin@42.fr>                    +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2019/09/28 14:47:25 by nkhribec          #+#    #+#             */
+/*   Updated: 2019/10/05 01:55:10 by nkhribec         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "header.h"
+
+char	*ft_realloc(char **s, size_t size)
+{
+	char	*tmp;
+
+	tmp = *s;
+	*s = ft_memalloc(sizeof(char) * size);
+	ft_strcpy(*s, tmp);
+	ft_strdel(&tmp);
+	return (*s);
+}
+
+void	nulltoblack(char *s, int until)
+{
+	while (until)
+	{
+		if (!*s)
+			*s = '^';
+		s++;
+		until--;
+	}
+}
+
+void	blacktozero(char *s, int until)
+{
+	while (until)
+	{
+		if (*s == '^')
+			*s = '0';
+		s++;
+		until--;
+	}
+}
+
+void	shift(char *s, int shift_value)
+{
+	int		i;
+
+	i = 0;
+	while (s[i])
+		i++;
+	while (--i >= 0)
+		s[i + shift_value] = s[i];
```

Comment:
shift_value = precision - len (len < precision)
and i have alredy allocate an str to take more than <precision_nbr> character.
so it will never cause a segfault.